pkg/storj/metainfo.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storj
+
+import (
+	"context"
+)
+
+// Metainfo represents a database for storing meta-info about objects
+type Metainfo interface {
+	// MetainfoLimits returns limits for this metainfo database
+	Limits() (MetainfoLimits, error)
+
+	// CreateBucket creates a new bucket with the specified information
+	CreateBucket(ctx context.Context, bucket Bucket) (Bucket, error)
```

Comment:
Shouldn't the `bucket` param here be of type `string`, i.e. the bucket name? It does not make sense for the caller to provide the Create time.

Reply:
Most likely the bucket will contain other methods not just the Create time.

I'm debating myself whether to get the "Bucket" name from the info struct or use a second argument for it.

The `CreateBucket` would override the `Created` time and `Modified`. (I'll add a comment about this.